File: src/main/java/testla/screenplay/action/Action.java
package testla.screenplay.action;

import testla.screenplay.actor.IActor;

/**
 * Actions can be triggered by calling them from an actor object.
 *
 * @author Patrick Doering
 */
public abstract class Action implements IAction {
    /**
     *  Makes the provided {@link IActor} perform this Action.
     *  This method will have to be overridden with the actual integration of an action.
     *
     * @param actor the IActor that will perform this action.
     * @return Object.
     */
    public abstract Object performAs(IActor actor);
}

File: src/main/kotlin/de/p7s1/testla/screenplay/Action.kt
package de.p7s1.testla.screenplay
import de.p7s1.testla.IAction
import de.p7s1.testla.IActor

abstract class Action: IAction {
    /**
    *  Makes the provided IActor perform this Action.
    *
    * @param actor the actor
    * @returns Any
    *
    * @override This method will have to be overridden with the actual integration of an action.
    */
    abstract override fun performAs(actor: IActor): Any
}